stock_picking_batch_oca/views/stock_batch_picking.xml

```diff
@@ -0,0 +1,185 @@
+                                    <field name="is_locked" invisible="1"/>
+                                    <field name="qty_done" attrs="{'readonly': [('state', 'in', ('done', 'cancel')), ('is_locked', '=', True)]}" force_save="1"/>
+                                </tree>
+                            </field>
+                        </page>
+                        <page string='Notes'>
+                            <field name='notes' colspan='4' nolabel='1'/>
+                        </page>
+                    </notebook>
+                </sheet>
+            </form>
+        </field>
+    </record>
+
+    <!-- stock.picking.batch tree view -->
+    <record model="ir.ui.view" id="stock_batch_picking_tree">
+        <field name="name">stock.picking.batch.tree</field>
+        <field name="model">stock.picking.batch</field>
+        <field name="arch" type="xml">
+            <tree string="Batch Picking">
+                <field name="name"/>
+                <field name="user_id"/>
+                <field name="date"/>
+                <field name="state"/>
+            </tree>
+        </field>
+    </record>
+
+    <record model="ir.actions.server" id="action_stock_batch_picking_assign_all">
+        <field name="name">Reserve All</field>
+        <field name="model_id" ref="stock_picking_batch.model_stock_picking_batch"/>
+        <field name="binding_model_id" ref="stock_picking_batch.model_stock_picking_batch"/>
+        <field name="state">code</field>
+        <field name="code">
+            batches = records.action_assign()
+        </field>
+    </record>
+
+    <!-- stock.picking.batch search view -->
+    <record id="stock_batch_picking_search" model="ir.ui.view">
+        <field name="name">batch.picking.search</field>
+        <field name="model">stock.picking.batch</field>
+        <field name="arch" type="xml">
+            <search>
+                <field name="name"/>
+                <field name="user_id"/>
+                <field name="state"/>
+                <field name="date"/>
+                <group expand="0" string="Group By...">
+                    <filter name="user_id" string="Picker" context="{'group_by':'user_id'}"/>
+                    <filter name="state" string="State" context="{'group_by':'state'}"/>
+                    <filter name="date" string="Order Date" context="{'group_by':'date'}"/>
+                </group>
+                <filter name="no_user_id" string="Unassigned" domain="[('user_id','=', False)]"/>
+            </search>
+        </field>
+    </record>
+
+    <record id="action_stock_batch_picking_tree"
+            model="ir.actions.act_window">
+        <field name="name">Batch Picking</field>
+        <field name="res_model">stock.picking.batch</field>
+        <field name="view_mode">tree,form</field>
+         <field name="view_id" ref="stock_batch_picking_tree"/>
+            <field name="view_ids" eval="[(5, 0, 0),
```

Comment:
Do we really need to remove existing related views here?
Also, not mandatory, but I think creating `act_window.view` separately is better to overload them easily (example: https://github.com/odoo/odoo/blob/8f4b98e67fe49516c0c830524de351db67c58357/addons/hr_timesheet/report/hr_timesheet_report_view.xml#L21) to change the sequence for instance.

Reply:
not removing now